core/include/byteorder.h

```diff
@@ -203,6 +203,48 @@ static inline uint32_t byteorder_ntohl(network_uint32_t v);
  */
 static inline uint64_t byteorder_ntohll(network_uint64_t v);
 
+/**
+ * @brief          Convert from host byte order to little endian , 16 bit.
+ * @param[in]      v   The integer in host byte order.
+ * @returns        `v` converted to little endian.
+ */
+static inline network_uint16_t byteorder_htols(uint16_t v);
```

Comment:
Here and basically everywhere: Type confusion ;-).

```suggestion
static inline le_uint16_t byteorder_htols(uint16_t v);
```